### PR TITLE
Improved GC metrics reporting

### PR DESCRIPTION
**Pull request by lutovich**
Replace usage of proprietary `com.sun.management.GarbageCollectionNotificationInfo` with plain `GarbageCollectorMXBean`. Added garbage collection count metric.

### CODE REVIEW

**Comment by chrisvest** [comment]
Nice. Looks good to me.

**Comment by benbc** [comment]
Looks good.

File: enterprise/metrics/src/main/java/org/neo4j/metrics/source/JvmMetrics.java [hunk 1]
```diff
@@ -23,45 +23,33 @@
...
 import org.neo4j.metrics.MetricsSettings;
 
 import static com.codahale.metrics.MetricRegistry.name;
-import static com.sun.management.GarbageCollectionNotificationInfo.GARBAGE_COLLECTION_NOTIFICATION;
 
 public class JvmMetrics extends LifecycleAdapter
 {
     private static final String NAME_PREFIX = "vm";
-    private static final String GC = name( NAME_PREFIX, "gc" );
+    private static final String GC_PREFIX = name( NAME_PREFIX, "gc" );
```
Comment (MishaDemianenko) [inline enterprise/metrics/src/main/java/org/neo4j/metrics/source/JvmMetrics.java:42]: I would assume we need to describe somewhere what those metrics show? Is it part of separate doc update commit?

**Comment by MishaDemianenko** [comment]
@lutovich small doc question otherwise well done!

File: enterprise/metrics/src/main/java/org/neo4j/metrics/source/JvmMetrics.java (hunk 1 above)
Comment (lutovich) [inline enterprise/metrics/src/main/java/org/neo4j/metrics/source/JvmMetrics.java:42]: Documentation will be available as part of separate PR. Here is the first step: https://github.com/neo4j/neo4j/pull/5793

**Comment by MishaDemianenko** [comment]
@lutovich needs rebase
